client/engine/engine.go

```diff
@@ -346,43 +384,73 @@ func (e *Engine) handleAPIEvent(apiEvent APIEvent) (ObjectiveChangeEvent, error)
+			if err != nil {
+				return EngineChangeEvent{}, fmt.Errorf("could not register channel with payment/receipt manager: %w", err)
 			}
 			return e.attemptProgress(&vfo)
 
 		case virtualdefund.ObjectiveRequest:
 			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
 			vdfo, err := virtualdefund.NewObjective(request, true, *e.store.GetAddress(), e.store.GetChannelById, e.store.GetConsensusChannel)
 			if err != nil {
-				return ObjectiveChangeEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
+				return EngineChangeEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
 			}
 			return e.attemptProgress(&vdfo)
 
 		case directfund.ObjectiveRequest:
 			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
 			dfo, err := directfund.NewObjective(request, true, *e.store.GetAddress(), e.store.GetChannelsByParticipant, e.store.GetConsensusChannel)
 			if err != nil {
-				return ObjectiveChangeEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
+				return EngineChangeEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
 			}
 			return e.attemptProgress(&dfo)
 
 		case directdefund.ObjectiveRequest:
 			e.metrics.RecordObjectiveStarted(request.Id(*e.store.GetAddress()))
 			ddfo, err := directdefund.NewObjective(request, true, e.store.GetConsensusChannelById)
 			if err != nil {
-				return ObjectiveChangeEvent{FailedObjectives: []protocols.ObjectiveId{request.Id(*e.store.GetAddress())}}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
+				return EngineChangeEvent{FailedObjectives: []protocols.ObjectiveId{request.Id(*e.store.GetAddress())}}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
 			}
 			// If ddfo creation was successful, destroy the consensus channel to prevent it being used (a Channel will now take over governance)
 			e.store.DestroyConsensusChannel(request.ChannelId)
 			return e.attemptProgress(&ddfo)
 
 		default:
-			return ObjectiveChangeEvent{}, fmt.Errorf("handleAPIEvent: Unknown objective type %T", request)
+			return EngineChangeEvent{}, fmt.Errorf("handleAPIEvent: Unknown objective type %T", request)
 		}
 
 	}
 
-	return ObjectiveChangeEvent{}, nil
+	zeroAddress := types.Destination{}
+	// TODO: Should this live in the payment manager?
+	if cId := apiEvent.PaymentToMake.ChannelId; cId != zeroAddress {
+		voucher, err := e.pm.Pay(
+			cId,
+			apiEvent.PaymentToMake.Amount,
+			*e.store.GetChannelSecretKey())
+		if err != nil {
+			return EngineChangeEvent{}, fmt.Errorf("handleAPIEvent: Error making payment: %w", err)
+		}
+		c, ok := e.store.GetChannelById(cId)
+		if !ok {
+			return EngineChangeEvent{}, fmt.Errorf("handleAPIEvent: Could not get channel from the store %s", cId)
+		}
+		sender, recipient := c.Participants[0], c.Participants[2]
```

Comment:
I think we also want to check that the channel is ready before sending the voucher. See https://github.com/statechannels/go-nitro/pull/800#discussion_r931904360 . I'm not sure the best place for that check to be, but I believe it is important in preventing race conditions -- essentially we want to be sure that the recipient has registered the channel in their receipt manager.

Reply:
> I think we also want to check that the channel is ready before sending the voucher. See [#800 (comment)](https://github.com/statechannels/go-nitro/pull/800#discussion_r931904360) . I'm not sure the best place for that check to be, but I believe it is important in preventing race conditions -- essentially we want to be sure that the recipient has registered the channel in their receipt manager.

Right now we use a crude solution to avoid race conditions, we register the virtual channel with the payment/receipt manager when we start creating the virtual channel. This means that we can rely on both participants having registered the virtual channel before any vouchers are sent.

This does mean that our payment/receipt manager will accept vouchers before the virtual channel is ready so we probably need to add a safeguard there.